crates/factor-outbound-networking/src/lib.rs

```diff
@@ -22,7 +22,24 @@ pub use runtime_config::ComponentTlsConfigs;
 
 pub type SharedFutureResult<T> = Shared<BoxFuture<'static, Result<Arc<T>, Arc<anyhow::Error>>>>;
 
-pub struct OutboundNetworkingFactor;
+#[derive(Default)]
+pub struct OutboundNetworkingFactor {
+    disallowed_host_callback: Option<DisallowedHostCallback>,
+}
+
+pub type DisallowedHostCallback = fn(scheme: &str, authority: &str);
```

Comment:
I know it's more typing, but should we only limit this to function pointers instead of any `impl Fn(&str, &str)`?

Reply:
Ugh but its _so much more typing_ :neutral_face: